llvm/lib/Target/AArch64/AArch64TargetMachine.cpp

```diff
@@ -894,7 +894,9 @@ void AArch64PassConfig::addPostBBSections() {
 void AArch64PassConfig::addPreEmitPass2() {
   // SVE bundles move prefixes with destructive operations. BLR_RVMARKER pseudo
   // instructions are lowered to bundles as well.
-  addPass(createUnpackMachineBundles(nullptr));
+  addPass(createUnpackMachineBundles([](const MachineFunction &MF) {
+    return MF.getFunction().hasFnAttribute("ct-select");
+  }));
```

Comment:
I think this will prevent the current bundle unpacking that's running on aarch64 due to the previous predicate function being a nullptr, except for the specific functions that have a ctselect. See [this](https://github.com/trail-of-forks/llvm-project/blob/ct-llvmorg-20.1.4/llvm/lib/CodeGen/MachineInstrBundle.cpp#L47-L48) check.